internal/report/report.go

```diff
@@ -28,6 +29,18 @@ type Writer struct {
 	minSeverity  config.Severity
 	showSeverity config.Severity
 	exclusions   []config.Exclusion
+	timeProvider TimeProvider
+}
+
+// TimeProvider provides a time server.
+type TimeProvider interface {
+	Now() time.Time
+}
+
+type realTime struct{}
+
+func (realTime) Now() time.Time {
+	return time.Now()
 }
```

Comment:
I would just create a local var that tests can set.

```suggestion
// timeNow is set by tests to mock the current time.
var timeNow = time.Now
```